### PR TITLE
Business and owner managment

### PR DESCRIPTION
**Pull request by ashkema**
Crazy stuff happened in this PR.
<ol>
<li>Created business management logic in `Shopipy.BusinessManagement`</li>
<li>Moved models, `AppDbContext`, and migrations to `Shopipy.Persistence`</li>
<li>Fixed a lousy cyclic include problem</li>
<li>Added `BusinessId` field to the User</li>
</ol>

### CODE REVIEW

File: Shopipy.ApiService/Controllers/AuthController.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
 using Shopipy.ApiService.DTOs;
-using Shopipy.UserManagement.Models;
+using Shopipy.Persistance.Models;
```
Comment (piatrashkakanstantinass) [inline Shopipy.ApiService/Controllers/AuthController.cs:5]: Persistence

File: Shopipy.BusinessManagement/Controllers/BusinessController.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+{
+    private readonly BusinessService _businessService;
+
+    public BusinessController(BusinessService businessService)
+    {
+        _businessService = businessService;
+    }
+    private BusinessResponseDto ToResponseDto(Business business) // main candidate for automapper
+    {
+        return new BusinessResponseDto
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/Controllers/BusinessController.cs:20]: Automapper geez

File: Shopipy.BusinessManagement/Controllers/BusinessController.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        var existingBusiness = await _businessService.GetBusinessByIdAsync(id);
+        if (existingBusiness == null)
+        {
+            return NotFound();
+        }
+
+        // Map fields
+        existingBusiness.Name = request.BusinessName;
+        existingBusiness.Address = request.BusinessAddress;
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/Controllers/BusinessController.cs:77]: That’s why I ask for automapper

File: Shopipy.BusinessManagement/DTOs/BusinessResponseDto.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+using System.Text.Json.Serialization;
+using Shopipy.Persistance.Models;
+
+namespace Shopipy.BusinessManagement.DTOs;
+
+public class BusinessResponseDto
+{
+    public int BusinessId { get; set; }
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/DTOs/BusinessResponseDto.cs:8]: Why int?

File: Shopipy.BusinessManagement/DTOs/BusinessResponseDto.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    public string Name { get; set; }
+    public string Address { get; set; }
+    public string Phone { get; set; }
+    public string Email { get; set; }
+    public string VATNumber { get; set; }
+    
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public BusinessType BusinessType { get; set; }
+
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/DTOs/BusinessResponseDto.cs:17]: ?

File: Shopipy.BusinessManagement/Repositories/BusinessRepository.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+using Shopipy.Persistance.Data;
+using Shopipy.Persistance.Models;
+
+namespace Shopipy.BusinessManagement.Repositories;
+
+public class BusinessRepository : IBusinessRepository
+{
+    private readonly AppDbContext _context;
+
+    public BusinessRepository(AppDbContext context)
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/Repositories/BusinessRepository.cs:12]: Use generic crudrepository, no one is going to copy paste same stuff

**Comment by piatrashkakanstantinass** [comment]
Fix property warnings, if property is not null add required keyword, otherwise mark it as nullable

File: Shopipy.Web/Views/_ViewImports.cshtml [hunk 7]
```diff
@@ -1,3 +1 @@
-﻿@using Shopipy.Web
-@using Shopipy.Web.Models
-@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
```
Comment (piatrashkakanstantinass) [inline Shopipy.Web/Views/_ViewImports.cshtml:3]: plz don't pollute the diff

File: Shopipy.Web/Program.cs [hunk 8]
```diff
@@ -1,3 +1,4 @@
+using Shopipy.ServiceDefaults;
```
Comment (piatrashkakanstantinass) [inline Shopipy.Web/Program.cs:1]: why was this added?

File: Shopipy.Web/Views/Shared/Error.cshtml [hunk 9]
```diff
@@ -1,4 +1,4 @@
-﻿@model ErrorViewModel
+﻿@model Shopipy.Web.Models.ErrorViewModel
```
Comment (piatrashkakanstantinass) [inline Shopipy.Web/Views/Shared/Error.cshtml:1]: why?

File: Shopipy.ServiceDefaults/Extensions.cs [hunk 10]
```diff
@@ -2,12 +2,13 @@
...
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
 using OpenTelemetry;
 using OpenTelemetry.Metrics;
 using OpenTelemetry.Trace;
 
-namespace Microsoft.Extensions.Hosting;
+namespace Shopipy.ServiceDefaults;
```
Comment (piatrashkakanstantinass) [inline Shopipy.ServiceDefaults/Extensions.cs:11]: this probably is the line that messed up some imports in Web project. Do not pollute the diff!

File: Shopipy.Persistance/Models/BusinessType.cs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+namespace Shopipy.Persistance.Models;
+
+public enum BusinessType
+{
+    Retail,
```
Comment (piatrashkakanstantinass) [inline Shopipy.Persistance/Models/BusinessType.cs:5]: did the write anything about business types and related handling? If not I would suggest dropping this part completely as it has no value.

File: Shopipy.BusinessManagement/DTOs/BusinessResponseDto.cs [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    public string Name { get; set; }
+    public string Address { get; set; }
+    public string Phone { get; set; }
+    public string Email { get; set; }
+    public string VATNumber { get; set; }
+    
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public BusinessType BusinessType { get; set; }
+    
+    public DateTime CreatedAt { get; set; }
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/DTOs/BusinessResponseDto.cs:18]: How db & our app is going to handle update of these 2 fields?

File: Shopipy.BusinessManagement/Controllers/BusinessController.cs [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace Shopipy.BusinessManagement.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class BusinessController : ControllerBase
+{
+    private readonly BusinessService _businessService;
+
+    public BusinessController(BusinessService businessService)
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/Controllers/BusinessController.cs:14]: use primary constructor

File: Shopipy.ApiService/Program.cs [hunk 14]
```diff
@@ -35,6 +39,9 @@
 
 builder.Services.AddAutoMapper(typeof(UserMappingProfile));
 
+builder.Services.AddScoped<IBusinessRepository, BusinessRepository>();
+builder.Services.AddScoped<BusinessService>();
```
Comment (piatrashkakanstantinass) [inline Shopipy.ApiService/Program.cs:43]: this logic belongs in business project, see [this](https://github.com/imbieras/shopipy/blob/kapi/authorization/Shopipy.UserManagement/DependencyInjection.cs)

File: Shopipy.Persistance/Models/BusinessType.cs (hunk 11 above)
Comment (ashkema) [reply Shopipy.Persistance/Models/BusinessType.cs:5]: business types are mentioned in the swagger, but no handling logic is provided

File: Shopipy.Persistence/Repositories/GenericRepository.cs [hunk 15]
```diff
@@ -0,0 +1,48 @@
+using Microsoft.EntityFrameworkCore;
+using Shopipy.Persistence.Data;
+
+namespace Shopipy.Persistence.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class
```
Comment (piatrashkakanstantinass) [inline Shopipy.Persistence/Repositories/GenericRepository.cs:6]: Cool stuff! 👌 Use primary constructors in your future PRs though.

File: Shopipy.BusinessManagement/Mappings/BusinessMappingProfile.cs [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        CreateMap<Business, BusinessResponseDto>()
+            .ForMember(dest => dest.BusinessId, opt => opt.MapFrom(src => src.BusinessId))
+            .ForMember(dest => dest.Name, opt => opt.MapFrom(src => src.Name))
+            .ForMember(dest => dest.Address, opt => opt.MapFrom(src => src.Address))
+            .ForMember(dest => dest.Phone, opt => opt.MapFrom(src => src.Phone))
+            .ForMember(dest => dest.Email, opt => opt.MapFrom(src => src.Email))
+            .ForMember(dest => dest.VatNumber, opt => opt.MapFrom(src => src.VatNumber))
+            .ForMember(dest => dest.BusinessType, opt => opt.MapFrom(src => src.BusinessType))
+            .ForMember(dest => dest.CreatedAt, opt => opt.MapFrom(src => src.CreatedAt))
+            .ForMember(dest => dest.UpdatedAt, opt => opt.MapFrom(src => src.UpdatedAt));
```
Comment (piatrashkakanstantinass) [inline Shopipy.BusinessManagement/Mappings/BusinessMappingProfile.cs:21]: The point of automapper is less work. If property names are the same automapper will do the mapping automatically. @ashkema

Comment (piatrashkakanstantinass) [reply Shopipy.BusinessManagement/Mappings/BusinessMappingProfile.cs:21]: Feel free to resolve this comment afterwards.

**Comment by piatrashkakanstantinass** [comment]
We probably need a task that combines user and business management (add user to business etc., since this part is related to authorization I can take it)

**Review by piatrashkakanstantinass — APPROVED** [review]
